### PR TITLE
[stdlib] Update the error msg to instruct using magic when compiler mismatches

### PR DESCRIPTION
**Pull request by msaelices**
It's still pointing to the deprecated modular CLI

### CODE REVIEW

File: stdlib/scripts/build-stdlib.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ if [ -z "${MOJO_OVERRIDE_COMPILER_VERSION_CHECK:-}" ]; then
     echo "Mismatch in compiler versions! Cannot build the standard library."
     echo "Expected compiler version: ${EXPECTED_COMPILER_VERSION}"
     echo "Current installed compiler version: ${ACTUAL_COMPILER_VERSION}"
-    echo "Please run modular update nightly/mojo to get the latest compiler."
+    echo "Please run 'magic update' and 'magic shell' to get the latest compiler."
```
Comment (gabrieldemarmiesse) [inline stdlib/scripts/build-stdlib.sh:30]: Maybe `magic update && magic install` to avoid having nested shells?

**Review by gabrieldemarmiesse — COMMENTED** [review]
I'm just wondering if this way to update is less error prone.

File: stdlib/scripts/build-stdlib.sh (hunk 1 above)
Comment (msaelices) [reply stdlib/scripts/build-stdlib.sh]: Agreed. Fixed here: https://github.com/msaelices/mojo/commit/20251d817a2923f3e0aec514cc5187e182c13426

**Review by JoeLoser — APPROVED** [review]

**Comment by JoeLoser** [comment]
!sync

**Pull request closed without merging**